src/aws/v4.rs

```diff
@@ -280,8 +289,8 @@ fn canonicalize_header(
             ctx.headers.insert(X_AMZ_DATE, date_header);
         }
 
-        // Insert X_AMZ_CONTENT_SHA_256 header if not present.
-        if ctx.headers.get(X_AMZ_CONTENT_SHA_256).is_none() {
+        // Insert X_AMZ_CONTENT_SHA_256 header if not present only for s3 request.
+        if ctx.headers.get(X_AMZ_CONTENT_SHA_256).is_none() && service == "s3" {
```

Comment:
This change doesn't LGTM, how about insert `X_AMZ_CONTENT_SHA_256` directly in request first.